### PR TITLE
nodes format validation

### PR DESCRIPTION
**Pull request by maha-hajja**
### Description

Related to https://github.com/conduitio-labs/conduit-connector-cassandra/pull/13#discussion_r1154694175

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-cassandra/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -43,5 +48,41 @@ func (d *DestinationConfig) validateConfig() error {
...
+		if err != nil {
+			return fmt.Errorf("invalid node format %q: %w", n, err)
+		}
+	}
+	return nil
+}
+
+func (d *DestinationConfig) validateHost(host string) error {
+	hostRegexRFC952 := regexp.MustCompile(`^[a-zA-Z]([a-zA-Z0-9\-]+[\.]?)*[a-zA-Z0-9]$`)
+	hostRegexRFC1123 := regexp.MustCompile(`^([a-zA-Z0-9]{1}[a-zA-Z0-9-]{0,62}){1}(\.[a-zA-Z0-9]{1}[a-zA-Z0-9-]{0,62})*?$`)
```
Comment (lovromazgon) [inline config.go:71]: We can move these into global variables, so we don't recompile the regex every time this function is called.
Also, both regexes seem to be very similar, we should probably choose one. Seems like the second one accepts more values (iiuc it allows numbers in front of the hostname), if Cassandra is fine with that we can just use that one.

File: config_test.go [hunk 2]
```diff
@@ -65,3 +65,79 @@ func TestConfig_AuthMechanism(t *testing.T) {
...
+func TestConfig_Nodes(t *testing.T) {
+	testCases := []struct {
+		name    string
+		config  DestinationConfig
+		wantErr bool
+	}{{
+		name: "port is greater than 65535",
+		config: DestinationConfig{
+			Nodes: []string{
+				"127.0.0.1:99999",
```
Comment (lovromazgon) [inline config_test.go:78]: Can we add a test for a hostname without a port? That's also allowed, right?

File: config.go (hunk 1 above)
Comment (maha-hajja) [reply config.go:71]: IIUC after doing more research, then compliance with RFC952 is a necessary but not sufficient condition for compliance with RFC1123. Also, an address can comply to RFC952 but not to RFC1123.
However, RFC1123 is generally considered the de facto standard for internet host requirements these days, so I'll go with that.

File: config_test.go (hunk 2 above)
Comment (maha-hajja) [reply config_test.go:78]: it's allowed by gocql yes, but wasn't sure if I should allow it to the user (lots of options and would have a longer description), but now that I'm thinking more about it, I think we should allow it yes.
I'm gonna have to do some validation changes to allow it first, then I'll add a test.

**Pull request merged**